components/script/dom/htmlslotelement.rs

```diff
@@ -364,9 +366,66 @@ impl Slottable {
                 }
             }
         }
+        println!("none 2");
         None
     }
 
+    /// Slottable name change steps from https://dom.spec.whatwg.org/#light-tree-slotables
+    pub(crate) fn update_name(&self, attr: &Attr, mutation: AttributeMutation) {
+        // Step 1. If localName is slot and namespace is null:
+        // NOTE: This is done by the caller
+        let old_value = if let AttributeMutation::Set(old_name) = mutation {
+            old_name.and_then(|attr| {
+                match &*attr {
+                    AttrValue::String(s) => Some(s.clone()),
+                    _ => None,
+                }
+            })
+        } else {
+            None
+        };
+        let value = mutation.new_value(attr).and_then(|attr| {
+            match &*attr {
+                AttrValue::String(s) => Some(s.clone()),
+                _ => None,
+            }
+        });
+
+        // Step 1.1 If value is oldValue, then return.
+        if value == old_value {
+            return;
+        }
+
+        // Step 1.2 If value is null and oldValue is the empty string, then return.
+        if value.is_none() && old_value.as_ref().is_some_and(|s| s.is_empty()) {
+            return;
+        }
+
+        // Step 1.3 If value is the empty string and oldValue is null, then return.
+        if old_value.is_none() && value.as_ref().is_some_and(|s| s.is_empty()) {
+            return;
+        }
+
+        // Step 1.4 If value is null or the empty string, then set element’s name to the empty string.
+        if !value.as_ref().is_some_and(|s| !s.is_empty()) {
```

Comment:
This conditional is surprisingly difficult for me to reason about. What about `value.as_ref().map_or(true, |s| s.is_empty())` instead?

Reply:
I just remembered [`Option::is_none_or`](https://doc.rust-lang.org/std/option/enum.Option.html#method.is_none_or) exists, that seems even nicer.